internal/service/advancedclustertpf/resource.go

```diff
@@ -99,80 +96,230 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
+			return "errorUpdateeAdvConfigLegacy", fmt.Sprintf(errorCreate, err.Error())
+		}
+	}
+
+	advConfigUpdate := NewAtlasReqAdvancedConfiguration(ctx, &plan.AdvancedConfiguration, diags)
+	var advConfig *admin20240805.ClusterDescriptionProcessArgs20240805
+	if advConfigUpdate != nil {
+		advConfig, _, err = api.UpdateClusterAdvancedConfiguration(ctx, projectID, clusterName, advConfigUpdate).Execute()
+		if err != nil {
+			// Maybe should be warning instead of error to avoid having to re-create the cluster
+			return "errorUpdateAdvConfig", fmt.Sprintf(errorCreate, err.Error())
+		}
+	}
+	return r.convertClusterAddAdvConfig(ctx, legacyAdvConfig, advConfig, cluster, plan.Timeouts, diags, &resp.State)
+}
+
+func (r *rs) readCluster(ctx context.Context, model *TFModel, state *tfsdk.State, diags *diag.Diagnostics) {
+	clusterName := model.Name.ValueString()
+	projectID := model.ProjectID.ValueString()
+	api := r.Client.AtlasV220240805.ClustersApi
+	readResp, _, err := api.GetCluster(ctx, projectID, clusterName).Execute()
 	if err != nil {
-		diags.AddError("errorCreate", fmt.Sprintf(errorCreate, err.Error()))
-		return nil, true
+		if admin.IsErrorCode(err, ErrorCodeClusterNotFound) {
+			state.RemoveResource(ctx)
+			return
+		}
+		diags.AddError("errorRead", fmt.Sprintf(errorRead, clusterName, err.Error()))
+		return
+	}
+	summary, detail := r.convertClusterAddAdvConfig(ctx, nil, nil, readResp, model.Timeouts, diags, state)
+	if summary != "" || detail != "" {
+		diags.AddError(summary, detail)
 	}
-	tfNewModel := NewTFModel(ctx, sdkResp, timeout, diags)
-	sdkAdvConfig, err := ReadClusterProcessArgsResponse()
+}
+func (r *rs) updateAndWait(ctx context.Context, patchReq *admin20240805.ClusterDescription20240805, diags *diag.Diagnostics, tfModel *TFModel) *admin20240805.ClusterDescription20240805 {
+	api := r.Client.AtlasV220240805.ClustersApi
+	projectID := tfModel.ProjectID.ValueString()
+	clusterName := tfModel.Name.ValueString()
+	_, _, err := api.UpdateCluster(ctx, projectID, clusterName, patchReq).Execute()
 	if err != nil {
-		diags.AddError("errorCreateAdvConfig", fmt.Sprintf(errorCreate, err.Error()))
-		return nil, true
+		diags.AddError("errorUpdate", fmt.Sprintf(errorUpdate, clusterName, err.Error()))
+		return nil
+	}
+	cluster, summary, detail := r.awaitChanges(ctx, &tfModel.Timeouts, diags, projectID, clusterName, "update")
+	if summary != "" || detail != "" {
+		diags.AddError(summary, detail)
+		return nil
 	}
-	sdkAdvConfigLegacy, err := ReadClusterProcessArgsResponseLegacy()
+	return cluster
+}
+
+func (r *rs) awaitChanges(ctx context.Context, t *timeouts.Value, diags *diag.Diagnostics, projectID, clusterName, changeReason string) (cluster *admin20240805.ClusterDescription20240805, summary, detail string) {
+	var timeoutDuration time.Duration
+	var localDiags diag.Diagnostics
+	var targetState = retrystrategy.RetryStrategyIdleState
+	var extraPending = []string{}
+	switch changeReason {
```

Comment:
we might want to use consts for changeReason values, also when calling awaitChange like:
```
// use const instead of "update"
cluster, summary, detail := r.awaitChanges(ctx, &tfModel.Timeouts, diags, projectID, clusterName, "update")
```

Reply:
Updated